Add link to add new carriers in the nav bar

Nada1996 opened the pull request:
<!--Read comments, before commiting pull request read checklist again

# Checklist:

- I have performed a self-review of my own code,
- I have commented my code, particularly in hard-to-understand areas,
- I have made corresponding changes to the documentation,
- I have added tests that prove my fix is effective or that my feature works,
- New and existing unit tests pass locally with my changes ("bundle exec rake"),
- Title include "WIP" if work is in progress.

-->

Resolves #110 

### Description
<!-- Please include a summary of the change and which issue is fixed. 
Please also include relevant motivation and context.
Guide questions:
  - What motivated this change (if not already described in an issue)?
  - What alternative solutions did you consider?
  - What are the tradeoffs for your solution?
   
List any dependencies that are required for this change. (gems, js libraries, etc.)

Include anything else we should know about. -->
I've adjusted the link of the `Add Item` icon in the navigation bar to link to `/carriers/new` to create a new carrier.

### Type of change

<!-- Please delete options that are not relevant. -->

* Bug fix (non-breaking change which fixes an issue)

### How Has This Been Tested?

<!-- Please describe the tests that you ran to verify your changes. 
Provide instructions so we can reproduce. 
Do we need to do anything else to verify your changes? 
If so, provide instructions (including any relevant configuration) so that 
we can reproduce? -->
I've added a new testcase that verify the current path in `carrier_spec.rb`. I'm also including a screenshot below.


### Screenshots
<!--Optional. Delete if not relevant. 
Include screenshots (before / after) for style changes, highlight 
edited element.-->
![Screenshot from 2019-10-13 20-24-15](https://user-images.githubusercontent.com/18237025/66720024-942a0e80-edf7-11e9-828b-adf64a7ca097.png)

Nada1996 committed:
add link to add new carriers in the nav bar

DamirSvrtan commented on app/views/layouts/_header.html.erb:
```diff
@@ -12,7 +12,7 @@
         </li>
         <li class="nav-item active background-slate">
-          <a class="nav-link nav-icon nav-second" href="/"><%= image_tag "BW- ADD USER ICON.svg", alt:"Add an Item" %><p>ADD ITEM</p></a></li>
+          <a class="nav-link nav-icon nav-second" href="/carriers/new"><%= image_tag "BW- ADD USER ICON.svg", alt:"Add an Item" %><p>ADD ITEM</p></a></li>
```
Please use the rails `link_to` syntax to link to other pages. You can see a couple of lines above how that is done for the signup path. In your case, it would be a `link_to new_carrier_path ....`

There are a couple of benefits of using rails helpers like the link_to syntax:

- if you really wanna change the url, you can change it through the routes.rb file and everything will work
- if for some reason we decide to remove the ability to add new carriers, or we change the link path all together, with the rails link_to syntax, rails will raise an exception and notify us about the link not linking properly anymore, while with a href link, it will just silently fail (the link will still be there but it won't link to a correct page)

DamirSvrtan reviewed (commented):
Thanks a lot for your contribution, I can't wait for the navigation to start working :-) I've added a comment with an explanation, so please just use the rails `link_to` syntax and i'll merge this right away!

Thanks again! 🎉

DamirSvrtan commented:
Hey @Nada1996 , as well could I please ask you to merge master into your own branch so that the specs can pass? We unfortunately had a master that had a failing spec and now all the PR's are showing failing Travis CI builds.

As well, it would be good to add a simple spec to test that the link you're going to add works (let me know if you need additional help on this).

Nada1996 committed:
Merge branch 'master' of git://github.com/rubyforgood/babywearing

Nada1996 committed:
Using link_to instead of anchor tag and adding a testcase scenario

Nada1996 commented:
> Hey @Nada1996 , as well could I please ask you to merge master into your own branch so that the specs can pass? We unfortunately had a master that had a failing spec and now all the PR's are showing failing Travis CI builds.
> 
> As well, it would be good to add a simple spec to test that the link you're going to add works (let me know if you need additional help on this).

Hi @DamirSvrtan , Thanks for reviewing the PR. I've changed it to use `link_to` and I added a new testcase to check if it's working properly.

DamirSvrtan commented:
Hey @Nada1996, perfect, thanks a lot! 🎉 Keep em coming and let me know if you need help picking up new issues!

Nada1996 commented:
Thanks @DamirSvrtan :smiley:
